src/main/java/io/kubernetes/KabaneroClient.java

```diff
@@ -96,35 +102,62 @@ private static ApiClient getApiClient() throws IOException, GeneralSecurityExcep
         return client;
     }
 
-    public static List<KabaneroInstance> getInstances() throws IOException, ApiException, GeneralSecurityException {
+    public static KabaneroList getInstances() throws IOException, ApiException, GeneralSecurityException {
         ApiClient client = KabaneroClient.getApiClient();
-
         String namespace = "kabanero";
+        try{
+            KabaneroApi api = new KabaneroApi(client);
+            KabaneroList kabaneros = api.listKabaneros(namespace, null, null, null);
+            List<Kabanero> kabaneroList = kabaneros.getItems();
 
-        List<KubeKabanero> instances = KabaneroClient.listKabaneroInstances(client, namespace);
-        LOGGER.log(Level.FINE, "Found {0} Kabanero Instances", instances.size());
-
-        List<KabaneroInstance> kabaneroInstances = new ArrayList<>();
-
-        for (KubeKabanero instance : instances) {
-
-            String username = null;
-            String instanceName = instance.getName();
-            String date = instance.getCreationTimestamp();
-            String cliURL = KabaneroClient.getCLI(client, namespace);
+			if (kabaneroList.size() > 0) {
+                return kabaneros;
+            }
+        }catch (Exception e) {
+			e.printStackTrace();
+		}
+        return null;
+    }
 
-            List<KabaneroRepository> kabaneroRepositories = instance.getRepositories();
-            List<KabaneroCollection> kabaneroCollections = KabaneroClient.listKabaneroCollections(client, namespace);
+    public static Kabanero getAnInstance(String instanceName) throws IOException, ApiException, GeneralSecurityException {
+        ApiClient client = KabaneroClient.getApiClient();
+        String namespace = "kabanero";
+        try{
+            KabaneroApi api = new KabaneroApi(client);
+            return api.getKabanero(namespace, instanceName);
+        }catch (Exception e){
+            e.printStackTrace();
+        }
+        return null;
+    }
 
-            String clusterName = null;
+    public static CollectionList getCollections(String instanceName) throws IOException, GeneralSecurityException {
+        ApiClient client = KabaneroClient.getApiClient();
+        try{
+            CollectionApi api = new CollectionApi(client);
+            CollectionList collections = api.listCollections(instanceName, null, null, null);
+            List<Collection> collectionList = collections.getItems();
+            if(collectionList.size() > 0){
+                return collections;
+            }
```

Comment:
no need to return only if size is greater than 0. Its okay to return an empty list